### PR TITLE
[Fabric] Shim RCTTextInputComponentView to work w/ Fabric

### PR DESCRIPTION
**Pull request by shwanton**
#### Please select one of the following
- [x] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [ ] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

Fabric on macOS implementation:
Shim RCTTextInputComponentView to work w/ Fabric

## Changelog

[macOS][Fabric] - Shim RCTTextInputComponentView for Fabric

## Test Plan

[x] Build RNTester-macOS w/ Fabric - doesn’t run yet, but no RCTTextInputComponentView errors
CleanShot 2022-12-05 at 17 51 06

Build errors: 
[Build RNTester-macOS_2022-12-05T17-49-32.txt](https://github.com/microsoft/react-native-macos/files/10160461/Build.RNTester-macOS_2022-12-05T17-49-32.txt)

[x] Build RNTester - iOS w/ Fabric - New architecture example should work
CleanShot 2022-12-05 at 18 00 49

[x] Build RNTester-macOS w/ Paper - should work
CleanShot 2022-12-05 at 18 12 31

[x] Build RNTester - iOS w/ Paper - should work
CleanShot 2022-12-05 at 18 15 32

### CODE REVIEW

File: React/Fabric/Mounting/ComponentViews/TextInput/RCTTextInputComponentView.mm [hunk 1]
```diff
@@ -556,12 +577,16 @@ - (void)_updateState
...
 {
+#if !TARGET_OS_OSX // [TODO(macOS GH#774)
   UITextRange *selectedTextRange = _backedTextInputView.selectedTextRange;
   NSInteger start = [_backedTextInputView offsetFromPosition:_backedTextInputView.beginningOfDocument
                                                   toPosition:selectedTextRange.start];
   NSInteger end = [_backedTextInputView offsetFromPosition:_backedTextInputView.beginningOfDocument
                                                 toPosition:selectedTextRange.end];
   return AttributedString::Range{(int)start, (int)(end - start)};
+#else
+  return AttributedString::Range({0, 1});
```
Comment (amgleitman) [inline React/Fabric/Mounting/ComponentViews/TextInput/RCTTextInputComponentView.mm:588]: If this is a placeholder, you should mark it as such and create an issue to implement it properly.

**Review by amgleitman — APPROVED** [review]

**Comment by amgleitman** [comment]
It's unfortunate that we can't test this component fully on Fabric macOS quite yet, but it makes sense given that it's still a work in progress. Here's hoping that day will come soon!

**Comment by shwanton** [comment]
@amgleitman very soon we'll be able to build fabric on macOS & then we should have a better way to test these components!

File: React/Fabric/Mounting/ComponentViews/TextInput/RCTTextInputComponentView.mm (hunk 1 above)
Comment (shwanton) [reply React/Fabric/Mounting/ComponentViews/TextInput/RCTTextInputComponentView.mm:589]: Added comment. Will add issue once I discuss best format for the Fabric follow up work w/
 @Saadnajmi

**Comment by shwanton** [comment]
closes https://github.com/microsoft/react-native-macos/issues/1553

**Pull request merged**